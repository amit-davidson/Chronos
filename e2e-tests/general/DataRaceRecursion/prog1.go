package main

import (
	"sync"
)

var a int
var mutex1 = sync.Mutex{}

func main() {
	go func() {
		a = 5
	}()
	fn1(0)
}

func fn1(counter int) {
	if counter >= 10 {
		return
	}
	if counter == 8 {
		a = 6
	}
	fn1(counter)
}
