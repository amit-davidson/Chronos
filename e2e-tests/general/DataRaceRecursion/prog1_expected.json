{
	"Lockset": {
		"ExistingLocks": {},
		"ExistingUnlocks": {}
	},
	"GuardedAccess": [
		{
			"ID": 1,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:4:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:4:11",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 1
				}
			}
		},
		{
			"ID": 2,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:4:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:4:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 2
				}
			}
		},
		{
			"ID": 3,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:5:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:5:11",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 3
				}
			}
		},
		{
			"ID": 4,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:5:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:5:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 4
				}
			}
		},
		{
			"ID": 5,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:7:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:5:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 1
				}
			}
		},
		{
			"ID": 6,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:7:4",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:7:3",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 2
				}
			}
		},
		{
			"ID": 7,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:7:4",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 3
				}
			}
		},
		{
			"ID": 8,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:8:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:4:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 2,
				"Clock": {
					"1": 5,
					"2": 4
				}
			}
		},
		{
			"ID": 9,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:10:2",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:5:2",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 6
				}
			}
		},
		{
			"ID": 10,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:10:3",
			"Value": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:10:2",
			"OpKind": 1,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 7
				}
			}
		},
		{
			"ID": 11,
			"Pos": "github.com/amit-davidson/Chronos/e2e-tests/general/DataRaceMap/prog1.go:10:3",
			"Value": "",
			"OpKind": 0,
			"LocksetJson": {
				"ExistingLocks": {},
				"ExistingUnlocks": {}
			},
			"State": {
				"GoroutineID": 1,
				"Clock": {
					"1": 8
				}
			}
		}
	]
}